src/seedu/addressbook/ui/Formatter.java

```diff
@@ -0,0 +1,99 @@
+package seedu.addressbook.ui;
+
+import static seedu.addressbook.common.Messages.MESSAGE_GOODBYE;
+import static seedu.addressbook.common.Messages.MESSAGE_INIT_FAILED;
+import static seedu.addressbook.common.Messages.MESSAGE_PROGRAM_LAUNCH_ARGS_USAGE;
+import static seedu.addressbook.common.Messages.MESSAGE_USING_STORAGE_FILE;
+import static seedu.addressbook.common.Messages.MESSAGE_WELCOME;
+
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Scanner;
+
+import seedu.addressbook.commands.CommandResult;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+import javax.swing.plaf.TextUI;
+
+
+/**
+ * Text UI of the application.
+ */
+public class Formatter {
+
+    /** A decorative prefix added to the beginning of lines printed by AddressBook */
+    static final String LINE_PREFIX = "|| ";
+
+    /** A platform independent line separator. */
+    static final String LS = System.lineSeparator();
+
+    static final String DIVIDER = "===================================================";
+
+    /** Format of indexed list item */
+    private static final String MESSAGE_INDEXED_LIST_ITEM = "\t%1$d. %2$s";
+
+
+    /** Offset required to convert between 1-indexing and 0-indexing.  */
+    public static final int DISPLAYED_INDEX_OFFSET = 1;
+
+    /** Format of a comment input line. Comment lines are silently consumed when reading user input. */
+    static final String COMMENT_LINE_FORMAT_REGEX = "#.*";
+
+
+    public Formatter() {
+
+    }
+
+    /**
+     * Shows the result of a command execution to the user. Includes additional formatting to demarcate different
+     * command execution segments.
+     */
+    public void showResultToUser(CommandResult result) {
```

Comment:
Should this be `Formatter` or `TextUi` responsibility? Furthermore each of the class is calling each others `showToUser` / `showResultToUser`

Reply:
I'm a little confused because showing results to the user would seem like a `TextUi` responsibility, however the idea of formatting the results into a presentable form is a `Formatter` responsibility. Should I put `showResultToUser` inside `TextUi` in such a case?

Reply:
Totally agree with you, showing a result is definitely the job of TextUi. Perhaps you can look from it this way. If you were to present a content regardless of what it is, it would be the job of `TextUi`; if you were to handle and format the content regardless of where it would be displayed, it would be the job of `Formatter`. 

It might be a good idea to return the result instead of calling `showToUser`